packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -0,0 +1,544 @@
+        username: props.masterUser.username,
+        encryptionKey: props.masterUser.encryptionKey,
+      });
+    }
+
+    // bind the engine to the Cluster
+    const clusterEngineBindConfig = props.engine.bindToCluster(this, {
+      parameterGroup: props.parameterGroup,
+    });
+    const clusterParameterGroup = props.parameterGroup ?? clusterEngineBindConfig.parameterGroup;
+    const clusterParameterGroupConfig = clusterParameterGroup?.bindToCluster({});
+
+    const cluster = new CfnDBCluster(this, 'Resource', {
+      backupRetentionPeriod: props.backupRetention?.toDays(),
+      databaseName: props.defaultDatabaseName,
+      dbClusterIdentifier: props.clusterIdentifier,
+      dbClusterParameterGroupName: clusterParameterGroupConfig?.parameterGroupName,
+      dbSubnetGroupName: this.subnetGroup.ref,
+      deletionProtection: props.deletionProtection,
+      engine: props.engine.engineType,
+      engineVersion: props.engine.engineVersion?.fullVersion,
+      engineMode: 'serverless',
+      enableHttpEndpoint: props.enableHttpEndpoint ?? false,
+      kmsKeyId: props.storageEncryptionKey?.keyArn,
+      masterUsername: secret ? secret.secretValueFromJson('username').toString() : props.masterUser.username,
+      masterUserPassword: secret
+        ? secret.secretValueFromJson('password').toString()
+        : (props.masterUser.password
+          ? props.masterUser.password.toString()
+          : undefined),
+      scalingConfiguration: props.scaling ? this.renderScalingConfiguration(props.scaling) : undefined,
+      storageEncrypted: true,
+      vpcSecurityGroupIds: this.securityGroups.map(sg => sg.securityGroupId),
+    });
+
+    this.clusterIdentifier = cluster.ref;
+
+    // create a number token that represents the port of the cluster
+    const portAttribute = Token.asNumber(cluster.attrEndpointPort);
+    this.clusterEndpoint = new Endpoint(cluster.attrEndpointAddress, portAttribute);
+    this.clusterReadEndpoint = new Endpoint(cluster.attrReadEndpointAddress, portAttribute);
+    this.connections = new ec2.Connections({
+      securityGroups: this.securityGroups,
+      defaultPort: ec2.Port.tcp(this.clusterEndpoint.port),
+    });
+
+    this.setRemovalPolicy(cluster, props.removalPolicy);
+
+    if (secret) {
+      this.secret = secret.attach(this);
+    }
+  }
+
+  /**
+   * Renders the secret attachment target specifications.
+   */
+  public asSecretAttachmentTarget(): secretsmanager.SecretAttachmentTargetProps {
+    return {
+      targetId: this.clusterIdentifier,
+      targetType: secretsmanager.AttachmentTargetType.RDS_DB_CLUSTER,
+    };
+  }
+
+  /**
+   * Adds the single user rotation of the master password to this cluster.
+   *
+   * @param [automaticallyAfter=Duration.days(30)] Specifies the number of days after the previous rotation
+   * before Secrets Manager triggers the next automatic rotation.
+   */
+  public addRotationSingleUser(automaticallyAfter?: Duration): secretsmanager.SecretRotation {
```

Comment:
If this logic is identical to cluster, maybe it can go into `private/util.ts`?

Reply:
yep, makes sense to me

Reply:
actually, i'll partially walk that back, there's also some duplicated logic between instance.
spoke offline: we'll do this refactor after we merge in #10313